script/main.go

```diff
@@ -17,44 +18,93 @@ import (
 	"golang.org/x/crypto/ssh/knownhosts"
 )
 
+type StringAndBool struct {
+	text string
+	boolValue bool
+}
+
+func getCommandInfo() map[string]StringAndBool {
+	// Map from command to help text and indication if sub-domain value is required (via flag).
+	return map[string]StringAndBool{
+		"build": {"Build the website, overwriting the selected-domain's file on server.", true},
+		"deploy": {"Build and upload the latest version of the website to selected sub-domain", true},
+		"preview": {"Start a local server for previewing the website.", false},
+		"upload": {"Upload the built website and host it at selected sub-domain.", true},
+		"rollback": {"Rollback the website to whatever was present before the last deploy.", true},
+		"rotatecert": {"Rotate the ssl (https) cert for supported domains.", false},
+		"listcerts": {"List all of the domains for which we can rotate certs.", false},
+		"listsubdomains": {"List all of the sub-domains we support generation of web pages.", false},
+		"help": {"Print this help text.", false},
+	}
+}
+
+
+func getSupportedSubDomains() map[string]StringAndBool {
+	// Map of sub-domain values in CLI to path to store files on server and if we support certs renewal.
+	return map[string]StringAndBool {
+		"simon": {"simon.duchastel.com", true},
+		"nicolas": {"nicolas.duchastel.com", true},
+		"pointbolin": {"pointbplin.com", true},
+		"com": {"duchastel.com", true},
+		"org": {"duchastel.org", true},
+	}
+}
+
+var command string
+var subDomain string
+
 func main() {
-	args := os.Args[1:] // ignore the first arg (program name)
+	flag.StringVar(&subDomain, "subdomain", "", "Directory where to put files on file server (for what sub-domain)")
+	flag.Parse()
+
+	args := flag.Args()
 	if len(args) != 1 {
-		fmt.Println("Error: Must provide exactly one command")
-		fmt.Println()
-		printHelp()
+		fmt.Println("Error: did not specify command. See help.")
```

Comment:
```suggestion
		fmt.Println("Error: did not specify command. Try 'help' command.")
```
Consistent with messaging on line 69, plus this might help make it more clear to use type the command `help`